### PR TITLE
Add rule to annotate GIHSN samples

### PR DESCRIPTION
**Pull request by joverlee521**
## Description of proposed changes

Adds a new `gihsn_sample` column to the metadata to indicate whether
"GIHSN" was found in the `strain` as proxy for whether the sample came
from the Global Influenza Hospital Surveillance Network (GIHSN).
Follows the existing pattern of using `True` and `False` boolean values.

## Related issue(s)

Resolves <https://github.com/nextstrain/seasonal-flu/issues/196>

## Checklist

- [x] Checks pass

### CODE REVIEW

File: workflow/snakemake_rules/select_strains.smk
Comment (joverlee521) [inline workflow/snakemake_rules/select_strains.smk:90]: I'm not tied to the `gihsn_sample` column name, so feel free to suggest any others.

**Review by huddlej — COMMENTED** [review]
Thank you for taking this one, @joverlee521! Seeing the implementation clarifies my original concerns about adding such a specific column to the metadata when the column represents <1% of all HA sequences and the same information exists in the strain name already.

Maybe we could figure out what the specific uses for this column would be before committing to this change? I can imagine two main use cases:

1. Users want to filter for sequences from the GIHSN in their subsampling logic for their builds, so they either can include those strains or exclude them. For this use case, users could use an argument like `--query "'GIHSN' in strain"` to select for those strains.
2. Users want to filter/color sequences in a tree by the GIHSN status. For this case, users would need an additional column in the metadata like the one from this PR to enable filtering/coloring in Auspice. If users are running their own builds, they could use a custom rule based on the same code in this PR to add the column to their subsampled metadata just before augur export runs (we could help them with this). If users want to apply these filters/colorings to our public builds, then we could use the same approach of adding the custom metadata column in the workflow just before export.

If the use cases are more about selecting/filtering sequences by sampling strategy, GISAID has a "sampling strategy" column in their metadata. It isn't populated for these GIHSN samples, but we could implement a derived version of that column for which we populate empty strings with "GIHSN", for example.

@rneher Are there other reasons we might need to have GIHSN as a column in our metadata for all samples?

**Comment by rneher** [comment]
I think having this as a column is probably easiest. This should compress very well and we are only talking about a hundred of thousand, rather than millions of rows.

**Comment by rneher** [comment]
I think the chosen column name is fine. this should help us to include them preferentially into the build and add a coloring.

**Comment by joverlee521** [comment]
Thanks both! I'll merge this so that we'll have the new column after Thursday's update.

File: workflow/snakemake_rules/select_strains.smk
Comment (genehack) [inline workflow/snakemake_rules/select_strains.smk:80]: do you want to `temp()` this?

**Review by genehack — APPROVED** [review]

File: workflow/snakemake_rules/select_strains.smk
Comment (joverlee521) [reply workflow/snakemake_rules/select_strains.smk:80]: (hah, you're comment popped up right when I clicked merge)

Yeah, adding `temp()` is probably best here.

Comment (genehack) [reply workflow/snakemake_rules/select_strains.smk:80]: > (hah, you're comment popped up right when I clicked merge)

Achievement unlocked: approve PR after it is already merged 🙌

Comment (joverlee521) [reply workflow/snakemake_rules/select_strains.smk:80]: Updated in https://github.com/nextstrain/seasonal-flu/pull/200

**Pull request merged**
